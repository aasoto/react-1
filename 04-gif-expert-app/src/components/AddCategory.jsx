import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState('')

  /** Cambia el estado del input
   * es decir que se cambia el texto que contiene
   */
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  }

  /** Se usa para evitar que el submit del formulario recargue la pÃ¡gina */
  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    // setCategories(categories => [inputValue, ...categories]);
    onNewCategory(inputValue.trim());
    setInputValue('');
  }

  return (
    <form onSubmit={(event => onSubmit(event))} aria-label="main-form">
      <input 
        type="text" 
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={ (event) => onInputChange(event) }
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}