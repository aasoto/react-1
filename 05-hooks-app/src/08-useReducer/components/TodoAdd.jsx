import { useState } from "react"
import { UseForm } from "../../hooks/UseForm";

export const TodoAdd = ({showTodo}) => {

  const {description, onInputChange, onResetForm} = UseForm({
      description: ''
    });

    const onFormSubmit = (event) => {
      event.preventDefault();

      if (description.length <= 1) return;

      const newTodo = {
        id: new Date().getTime(),
        done: false,
        description: description,
      }

      showTodo(newTodo);
      onResetForm();
    }

  // const [formState, setFormState] = useState({
  //   id: new Date().getTime(),
  //   description: '',
  //   done: false
  // });

  // const {description} = formState;

  // const onInputChange = ({target}) => {
  //   const {name, value} = target;
  //   setFormState({
  //     ...formState,
  //     [ name ]: value,
  //   });
  // }

  // const onSubmit = (event) => {
  //   event.preventDefault();
  //   showTodo(formState);
  // }

  return (
    <form onSubmit={onFormSubmit}>
      <input 
        type="text" 
        placeholder="¿Qué hay que hacer?"
        className="form-control"
        name="description"
        value = {description}
        onChange = {(event) => onInputChange(event)}
      />

      <button
        type="submit"
        className="btn btn-outline-primary mt-2"
      >
        Agregar
      </button>
    </form>
  )
}
