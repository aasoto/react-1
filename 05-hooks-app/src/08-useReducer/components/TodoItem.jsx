
export const TodoItem = ({todo, onDeleteTodo}) => {
  const {id, description} = todo;
  return (
    <li key={id} className="list-group-item d-flex justify-content-between">
      <span className="align-self-center">{description}</span>
      <button 
        className="btn btn-danger"
        onClick={() => onDeleteTodo(id)}
      >
        Borrar
      </button>
    </li>
  )
}
