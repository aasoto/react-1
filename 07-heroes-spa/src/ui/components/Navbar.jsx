import { useContext } from 'react';
import { Link, NavLink, useNavigate } from 'react-router-dom';
import { AuthContext } from '../../auth';


export const Navbar = () => {

  /** Este es un custom Hook de react-router-dom para facilitar la navegación
   * entre componentes.
   * Los Hooks que son hechos por librería de terceros como en esta caso react-router-dom tambien se conocen como custom Hooks.
   */
  const navigate = useNavigate();

  /** Se obtiene las variables y funciones del contexto de la autentición */
  const { user, logout } = useContext(AuthContext);

  const onLogout = () => {
    logout();
    /** Cuando se cierra la sesión se debe reemplazar la ruta para que no se puedan regresar */
    navigate('/login', {
      replace: true
    });
  }

  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark p-2">

      <Link
        className="navbar-brand"
        to="/"
      >
        Asociaciones
      </Link>

      <div className="navbar-collapse">
        <div className="navbar-nav">

          <NavLink
            className={ ({isActive}) => `nav-item nav-link ${ isActive ? 'active' : '' }` }
            to="/marvel"
          >
            Marvel
          </NavLink>

          <NavLink
            className={ ({isActive}) => `nav-item nav-link ${ isActive ? 'active' : '' }` }
            to="/dc"
          >
            DC
          </NavLink>
          <NavLink
            className={ ({isActive}) => `nav-item nav-link ${ isActive ? 'active' : '' }` }
            to="/search"
          >
            Search
          </NavLink>
        </div>
      </div>

      <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end">
        <ul className="navbar-nav ml-auto">
          
          <span className='nav-item nav-link text-primary'>
            {user?.name}
          </span>

          <button 
            className='nav-item nav-link btn'
            aria-label='btn-logout'
            onClick={() => onLogout()}
          >
            Logout
          </button>
        </ul>
      </div>
    </nav>
  )
}