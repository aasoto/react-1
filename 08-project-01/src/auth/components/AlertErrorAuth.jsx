
export const AlertErrorAuth = ({show = false}) => {
  return (
    <div className={`w-full h-11 bg-red-200 border border-red-600 text-red-600 flex justify-center items-center ${show ? 'block' : 'hidden'}`}>
      <h4>The email and/or password are wrong</h4>
    </div>
  )
}
