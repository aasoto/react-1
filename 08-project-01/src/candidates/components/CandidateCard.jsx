
export const CandidateCard = () => {
  return (
    <div>CandidateCard</div>
  )
}
