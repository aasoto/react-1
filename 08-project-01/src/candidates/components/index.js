export * from './CandidateCard';
export * from './CandidateList';
export * from './SearchCandidateInput';