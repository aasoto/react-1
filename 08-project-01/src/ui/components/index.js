export * from './Navbar';
export * from './OptionsNavbar';
export * from './Sidebar';