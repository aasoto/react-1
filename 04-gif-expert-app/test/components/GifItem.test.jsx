import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas de <GifItem/>', () => {

  const image = {
    id: 1,
    title: 'imagen',
    url: 'http://gifexpect.xyz/images/123.jpg'
  };

  test('hace match con el snapshot', () => {

    const {container} = render( <GifItem key={image.id} {...image}/>);
    expect(container).toMatchSnapshot();
  });

  test('mostrar URL de la imagen y el ALT indicado', () => {

    render(<GifItem title={image.title} url={image.url}/>);

    const {src, alt} = screen.getByRole('img');
    expect(src).toBe(image.url);
    expect(alt).toBe(image.title);
  });

  test('mostrar el tÃ­tulo en el componente', () => {

    render(<GifItem title={image.title} url={image.url}/>);
    expect(screen.getByText(image.title)).toBeTruthy();

  })

});
