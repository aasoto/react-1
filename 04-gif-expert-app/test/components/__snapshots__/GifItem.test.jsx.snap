// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifItem/> hace match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="imagen"
      src="http://gifexpect.xyz/images/123.jpg"
    />
    <p>
      imagen
    </p>
  </div>
</div>
`;
